packages/config/src/index.ts

```diff
@@ -186,12 +186,29 @@ export default async (
     'workspace-concurrency': 4,
     'workspace-prefix': opts.workspaceDir,
   })
+
+  const settingKeys = [
+    ...Object.keys(npmConfig?.sources?.project?.data ?? {}),
+    ...Object.keys(npmConfig?.sources?.workspace?.data ?? {}),
+    ...Object.keys(npmConfig?.sources?.user?.data ?? {}),
+  ]
+  const rcOptions = Object.keys(rcOptionsTypes)
+  const unknownKeys = []
+  for (const key of settingKeys) {
+    if (!rcOptions.includes(key)) {
+      unknownKeys.push(key)
+    }
+  }
+  if (unknownKeys.length) {
+    warnings.push(`Your .npmrc file contains unknown setting: ${unknownKeys.join(', ')}`)
+  }
```

Comment:
I am afraid there will be too many warnings. Might be good, for now, to report this warnings only when useBetaCli is true. Also, seems like this new code can be moved to a function

Reply:
Arguably, this should only run for `.pnpmrc`: <https://github.com/pnpm/pnpm/issues/2803>